mixer/pkg/protobuf/yaml/encoder.go

```diff
@@ -63,69 +63,909 @@ func (e *Encoder) EncodeBytes(data map[interface{}]interface{}, msgName string,
 }
 
 func (e *Encoder) visit(name string, data interface{}, field *descriptor.FieldDescriptorProto, skipUnknown bool, buffer *proto.Buffer) error {
+	if data == nil {
+		return nil
+	}
+	repeated := field.IsRepeated()
+	packed := field.IsPacked() || field.IsPacked3()
+	wireType := uint64(field.WireType())
+	fieldNumber := int(field.GetNumber())
+	if packed {
+		wireType = uint64(proto.WireBytes)
+	}
 	switch field.GetType() {
-	case descriptor.FieldDescriptorProto_TYPE_STRING:
-		v, ok := data.(string)
-		if !ok {
-			return badTypeError(name, "string", data)
+	case descriptor.FieldDescriptorProto_TYPE_DOUBLE:
+		if packed {
+			/*
```

Comment:
Consider using "//" for commenting. That works better when block-commenting is used for temporary code disablement.

Reply:
D